webapp/src/app/shared/widgets/string-widget/string-widget.component.ts

```diff
@@ -23,5 +28,34 @@ export class StringWidgetComponent extends AbstractWidget implements OnInit {
 				this.keys.unshift('');
 			}
 		}
+		this.updateSuggestedOptions();
+	}
+	
+	updateSuggestedOptions() {
+		const inputText = this.settings[this.key];
+		const searchResults = this.searchFilterService.filterOptions(this.keys, inputText);
+		
+		if (searchResults.some(option => option.length === inputText.length)) {
```

Comment:
`inputText` may be null or undefined. I would rather have an exact check or a `.startsWith` here anyway since it can lead to confusing ordering otherwise.